aten/src/ATen/native/cuda/Loops.cuh

```diff
@@ -102,20 +102,18 @@ static void launch_kernel(int64_t N, const func_t& f) {
   AT_CUDA_CHECK(cudaGetLastError());
 }
 
-template <typename traits, typename index_t, std::size_t... I>
-C10_HOST_DEVICE typename traits::ArgsTuple
-dereference_impl(char* const C10_RESTRICT data[], const index_t strides[], int i,
-                 c10::guts::index_sequence<I...>) {
-  return std::make_tuple(
-      *(typename traits::template arg<I>::type*)
-        (data[I] + i * strides[I])...);
+template <typename traits, typename func_t, typename index_t, size_t... I>
+C10_HOST_DEVICE typename traits::result_type
+invoke_impl(const func_t &f, char *const data[], const index_t strides[], int i,
```

Comment:
Was removing `C10_RESTRICT` necessary?

IIRC, the restrict improved the performance for some kernels.

Reply:
mistake..

I will test with `C10_RESTRICT`. And push it.